### PR TITLE
CollectionSystemManager: Fix possible crash in destructor

### PR DESCRIPTION
**Pull request by sylt**
This could happen if no mCustomCollectionsBundle was initialized.

### CODE REVIEW

**Comment by Gemba** [comment]
Can you pls provide evidence in which scenario(s) mCustomCollectionBundle is not initialized (ie. ES crashes)?

**Comment by pjft** [comment]
Apologies, I left this PR on my inbox. In principle, I like "safer" code, so I don't object to the change - thanks for looking into it.

I am, however, curious about when would this happen. From reading the code, it seems that the only option for not initializing that object is if there's an issue with the es_systems.cfg . Is that the case and, if that happens, does it effectively crash there?

Thanks both for the comments here.

**Comment by sylt** [comment]
Ah, my apologies for not providing enough information! Let me try again.

As I was trying out ES for the first time on my computer (I normally run it on my RPi), this was a crash I ran into when I had a non-configured system. Using latest master (b5e195050), this is the way I reproduce the crash:

0. Delete your ~/.emulationstation folder.
1. Start up emulationstation.
2. When ES has finished initializing, a pop-up saying: "IT LOOKS LIKE YOUR SYSTEM CONFIGURATION FILE HAS NOT BEEN SETUP OR IS INVALID".
3. Press F4 to quit.
4. :boom: (SIGSEGV)

The log from the run is as provided below, along with a stack trace of the crash:

```
Creating config directory "/home/dinho/.emulationstation"
Aug 27 17:43:51 lvl2: 	EmulationStation - v2.11.0rp-dev, built Aug 27 2022 - 17:42:50
Aug 27 17:43:51 lvl2: 	Parsing XML file "/home/dinho/code/EmulationStation/resources/mamenames.xml"...
Aug 27 17:43:51 lvl2: 	Parsing XML file "/home/dinho/code/EmulationStation/resources/mamebioses.xml"...
Aug 27 17:43:51 lvl2: 	Parsing XML file "/home/dinho/code/EmulationStation/resources/mamedevices.xml"...
Aug 27 17:43:51 lvl2: 	Creating window...
Aug 27 17:43:52 lvl2: 	Created window successfully.
Aug 27 17:43:52 lvl2: 	GL vendor:   NVIDIA Corporation
Aug 27 17:43:52 lvl2: 	GL renderer: NVIDIA GeForce GTX 660/PCIe/SSE2
Aug 27 17:43:52 lvl2: 	GL version:  4.6.0 NVIDIA 470.141.03
Aug 27 17:43:52 lvl2: 	Checking available OpenGL extensions...
Aug 27 17:43:52 lvl2: 	 ARB_texture_non_power_of_two: ok
Aug 27 17:43:52 lvl2: 	Loading system config file /etc/emulationstation/es_systems.cfg...
Aug 27 17:43:52 lvl0: 	es_systems.cfg file does not exist!
Aug 27 17:43:52 lvl0: 	Example config written!  Go read it at "/home/dinho/.emulationstation/es_systems.cfg"!
Aug 27 17:43:52 lvl0: 	Error while parsing systems configuration file!
Aug 27 17:43:52 lvl2: 	Added unconfigured joystick 'Xbox One S Controller' (GUID: 030000005e040000ea02000001030000, instance ID: 0, device index: 0).
Aug 27 17:43:52 lvl2: 	Added unconfigured joystick 'Xbox One S Controller' (GUID: 030000005e040000ea02000001030000, instance ID: 0, device index: 0).

Thread 1 "emulationstatio" received signal SIGSEGV, Segmentation fault.
0x00005555558079b6 in SystemData::getRootFolder (this=0x0) at /home/dinho/code/EmulationStation/es-app/src/SystemData.h:33
33		inline FileData* getRootFolder() const { return mRootFolder; };
(gdb) bt
#0  0x00005555558079b6 in SystemData::getRootFolder (this=0x0) at /home/dinho/code/EmulationStation/es-app/src/SystemData.h:33
#1  0x00005555558295ec in CollectionSystemManager::removeCollectionsFromDisplayedSystems (this=0x555555b87e20) at /home/dinho/code/EmulationStation/es-app/src/CollectionSystemManager.cpp:817
#2  0x00005555558230c6 in CollectionSystemManager::~CollectionSystemManager (this=0x555555b87e20, __in_chrg=<optimized out>)
    at /home/dinho/code/EmulationStation/es-app/src/CollectionSystemManager.cpp:67
#3  0x00005555558232cd in CollectionSystemManager::deinit () at /home/dinho/code/EmulationStation/es-app/src/CollectionSystemManager.cpp:97
#4  0x00005555557fa54b in main (argc=2, argv=0x7fffffffe678) at /home/dinho/code/EmulationStation/es-app/src/main.cpp:473
``` 

As is possible to see from the above, we will take the code path in `SystemData::loadConfig()` which emits the trace `es_systems.cfg file does not exist!`. So, the method will return false early on, meaning that we will never reach the call to `CollectionSystemManager::get()->loadCollectionSystems()`, which is the method that initializes `mCustomCollectionsBundle`. Finally, as we call `CollectionSystemManager::deinit()` when it's time to quit, we will find ourselves in a situation where `mCustomCollectionsBundle` is NULL.

My proposed fix to this was simply to check of `mCustomCollectionsBundle` was NULL or not, since it seemed simple enough.

Please let me know if there's any more information I can provide!

**Comment by pjft** [comment]
Makes sense - just wanted to confirm how you got there :) 

Thanks for sending this over!

**Comment by Gemba** [comment]
Nice catch @sylt :+1: and thanks for the scenario.
